Mathlib/Data/TypeVec.lean

```diff
@@ -0,0 +1,862 @@
+  :=
+  (splitFun_comp _ _ _ _).symm
+#align typevec.append_fun_comp_split_fun TypeVec.appendFun_comp_splitFun
+
+theorem appendFun_comp  {α₀ α₁ α₂ : TypeVec n}
+                        {β₀ β₁ β₂ : Type _}
+                        (f₀ : α₀ ⟹ α₁) (f₁ : α₁ ⟹ α₂)
+                        (g₀ : β₀ → β₁) (g₁ : β₁ → β₂) :
+    (f₁ ⊚ f₀ ::: g₁ ∘ g₀) = (f₁ ::: g₁) ⊚ (f₀ ::: g₀) :=
+  eq_of_drop_last_eq rfl rfl
+#align typevec.append_fun_comp TypeVec.appendFun_comp
+
+theorem appendFun_comp' {α₀ α₁ α₂ : TypeVec n} {β₀ β₁ β₂ : Type _}
+                        (f₀ : α₀ ⟹ α₁) (f₁ : α₁ ⟹ α₂)
+                        (g₀ : β₀ → β₁) (g₁ : β₁ → β₂) :
+    (f₁ ::: g₁) ⊚ (f₀ ::: g₀) = (f₁ ⊚ f₀ ::: g₁ ∘ g₀) :=
+  eq_of_drop_last_eq rfl rfl
+#align typevec.append_fun_comp' TypeVec.appendFun_comp'
+
+theorem nilFun_comp {α₀ : TypeVec 0} (f₀ : α₀ ⟹ Fin2.elim0) : nilFun ⊚ f₀ = f₀ :=
+  funext fun x => by apply Fin2.elim0 x -- porting note: `by apply` is necessary?
+#align typevec.nilFun_comp TypeVec.nilFun_comp
+
+theorem appendFun_comp_id {α : TypeVec n} {β₀ β₁ β₂ : Type _} (g₀ : β₀ → β₁) (g₁ : β₁ → β₂) :
+    (@id _ α ::: g₁ ∘ g₀) = (id ::: g₁) ⊚ (id ::: g₀) :=
+  eq_of_drop_last_eq rfl rfl
+#align typevec.append_fun_comp_id TypeVec.appendFun_comp_id
+
+@[simp]
+theorem dropFun_comp {α₀ α₁ α₂ : TypeVec (n + 1)} (f₀ : α₀ ⟹ α₁) (f₁ : α₁ ⟹ α₂) :
+    dropFun (f₁ ⊚ f₀) = dropFun f₁ ⊚ dropFun f₀ :=
+  rfl
+#align typevec.drop_fun_comp TypeVec.dropFun_comp
+
+@[simp]
+theorem lastFun_comp {α₀ α₁ α₂ : TypeVec (n + 1)} (f₀ : α₀ ⟹ α₁) (f₁ : α₁ ⟹ α₂) :
+    lastFun (f₁ ⊚ f₀) = lastFun f₁ ∘ lastFun f₀ :=
+  rfl
+#align typevec.last_fun_comp TypeVec.lastFun_comp
+
+theorem appendFun_aux {α α' : TypeVec n} {β β' : Type _} (f : (α ::: β) ⟹ (α' ::: β')) :
+    (dropFun f ::: lastFun f) = f :=
+  eq_of_drop_last_eq rfl rfl
+#align typevec.append_fun_aux TypeVec.appendFun_aux
+
+theorem appendFun_id_id {α : TypeVec n} {β : Type _} :
+    (@TypeVec.id n α ::: @_root_.id β) = TypeVec.id :=
+  eq_of_drop_last_eq rfl rfl
+#align typevec.append_fun_id_id TypeVec.appendFun_id_id
+
+instance subsingleton0 : Subsingleton (TypeVec 0) :=
+  ⟨fun a b => funext fun a => by apply Fin2.elim0 a⟩ -- porting note: `by apply` necessary?
+#align typevec.subsingleton0 TypeVec.subsingleton0
+
+-- /- porting note: just omitting this `simp` attribute declaration for now
+
+-- -/
+
+
+/-- cases distinction for 0-length type vector -/
+protected def casesNil {β : TypeVec 0 → Sort _} (f : β Fin2.elim0) : ∀ v, β v
+  := fun v => cast (by congr; funext i; cases i) f
+#align typevec.cases_nil TypeVec.casesNil
+
+/-- cases distinction for (n+1)-length type vector -/
+protected def casesCons (n : ℕ) {β : TypeVec (n + 1) → Sort _}
+                        (f : ∀ (t) (v : TypeVec n), β (v ::: t)) :
+    ∀ v, β v :=
+  fun v : TypeVec (n + 1) => cast (by simp) (f v.last v.drop)
+#align typevec.cases_cons TypeVec.casesCons
+
+protected theorem cases_nil_append1 {β : TypeVec 0 → Sort _} (f : β Fin2.elim0) :
```

Comment:
```suggestion
protected theorem casesNil_append1 {β : TypeVec 0 → Sort _} (f : β Fin2.elim0) :
```

Reply:
Thanks for catching, I fixed the naming!

A question about procedure, should I resolve a conversation when I apply the suggested fix to get them out of the way, or is it best to leave them for the reviewer, so they can check I indeed fixed it as intended?

Reply:
In the case of naming suggestions, please resolve yourself. If there's more of a discussion going on, then please leave it open until both sides agree that the discussion is resolved.